[PLATFORM-992]: Ensure alignment of traces semantic convention

cpiemontese opened the pull request:
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-995

This PR adds the [opentelemetry_semantic_conventions](https://hexdocs.pm/opentelemetry_semantic_conventions/0.2.0/) library as a dependency to ensure we follow the proper convention names

MaeIsBad commented on mix.exs:
```diff
@@ -2,7 +2,7 @@ defmodule Telepoison.MixProject do
 
   @source_url "https://github.com/primait/telepoison"
-  @version "1.1.2"
+  @version "1.2.0"
```
I don't think we should bump the package version as a part of this PR?
Also this seems like a minor change, since I don't think it's even observable from the end-user perspective

cpiemontese replied:
I bumped it because of a previous attempt to release but it doesn't make sense, I reverted the bump

MaeIsBad reviewed (changes requested):
Small nitpick: there is a mention of `http.host` in the readme file
https://github.com/primait/telepoison/tree/PLATFORM-995/user-story/use-otel-recommended-names-for-attributes#how-it-works

cpiemontese commented:
Not small at all, I didn't think this was documented! I'll fix the README

MaeIsBad reviewed (commented):
another thing I just noticed: we don't follow the otel recommendations related to the span name

>HTTP client spans have no http.route attribute since client-side instrumentation is not generally aware of the “route”, and therefore HTTP client spans SHOULD use {http.method}. Instrumentation MUST NOT default to using URI path as span name, but MAY provide hooks to allow custom logic to override the default span name.

Our spans use
```
  def compute_default_span_name(request) do
    method_str = request.method |> Atom.to_string() |> String.upcase()
    %URI{authority: authority} = request.url |> process_request_url() |> URI.parse()
    "#{method_str} #{authority}"
  end
```

cpiemontese commented:
Thanks for the thorough review, I kinda made this PR with my feet :laughing: 

I changed the default span name to be just the method (and also updated a test to check for it)
I updated the README so that it no longer references `http.host`
